cli/src/commands/create/comments.rs

```diff
@@ -40,9 +41,42 @@ pub struct CreateCommentsArgs {
     /// Number of comments to batch in a single request.
     batch_size: usize,
 
-    #[structopt(long = "progress")]
-    /// Whether to display a progress bar. Only applicable when a file is used.
-    progress: Option<bool>,
+    #[structopt(long)]
+    /// Whether to display a progress bar (only applicable when --file is used).
+    progress: Option<Option<bool>>,
```

Comment:
Yes precisely as you say - there's 4 states! `(unset)`, `--flag`, `--flag=true`, `--flag=false`.

You could argue that the true/false split is unnecessary and there's just `(unset)` and `--flag`.

But for `--progress`, the current default is `true` and so the only "meaningful" value there is `--progress=false`.

Proposal: this can all get simplified by making none of these accept `=true/=false`, and instead make the progress flag `--no-progress`. Then the args would be:

```
--no-progress        (Don't show progress when uploading a file.)
--allow-duplicates  (Allow duplicates in the upload source.)
--overwrite             (Allow overwriting conflicting comments.)
```

Reply:
One reason why allowing values is that it makes scripts somewhat more ergonomic to write as I can have a variable `VALUE` which is `true` or `false` and write `--argument=$VALUE` without having to conditionally build a string and append (or not) the flag.

Having said this, let's go with your proposal :+1: